File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        for (int i = 1; i < nums.size(); i++) {
            nums[i] = max(i + nums[i], nums[i - 1]);
        }

        int jumps = 0;
        int idx = 0;
        while (idx < nums.size() - 1) {
            jumps++;
            idx = nums[idx];
        }

        return jumps;
    }
};

File: top-interview-150/0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        dp = []
        dp.append(nums[0])
        for i in range(1, len(nums)):
            dp.append(max(i + nums[i], dp[i - 1]))

        jumps = 0
        idx = 0
        while idx <= len(nums) - 2:
            jumps += 1
            idx = dp[idx]

        return jumps